terraform/cloudfront/response-immutable.js

```diff
@@ -12,6 +12,7 @@ function handler(event) {
   headers['x-content-type-options'] = { value: 'nosniff' };
   headers['x-frame-options'] = { value: 'DENY' };
   headers['x-xss-protection'] = { value: '1; mode=block' };
+  headers['cache-control'] = { value: 'public, max-age=604800, immutable' }
```

Comment:
How do the two response files work together and why are there two different `cache-control` headers?

Reply:
Response functions are attached to behaviours (`ordered_cache_behavior`). Behaviours are applied to matching regexs on the URL. So different URL patterns can have different functions. I have mapped urls with `_next` to be cached forever (since they all have cache busting URLs) and otherwise the default behaviour is no caching.

Reply:
Ahh so the `_next` directory will be cached _forever_ up until we deploy new changes?

Reply:
Yeah, then on a new deploy the index.html (uncached) will pull different .js/.css files (with different cache busting URLs)